x/market/keeper/deal_test.go

```diff
@@ -255,6 +255,108 @@ func (suite *dealTestSuite) TestVerify() {
 	suite.Require().NoError(err)
 }
 
+func (suite *dealTestSuite) TestDeactivateDeal() {
+	err := suite.BankKeeper.AddCoins(suite.Ctx, acc1, sdk.NewCoins(sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(10000000000))))
+	suite.Require().NoError(err)
+
+	err = suite.BankKeeper.AddCoins(suite.Ctx, acc3, zeroFunds)
+	suite.Require().NoError(err)
+
+	tempDeal := types.Deal{
+		DataSchema:            []string{"http://jsonld.com"},
+		Budget:                &sdk.Coin{Denom: assets.MicroMedDenom, Amount: sdk.NewInt(1000000000)},
+		MaxNumData:            10000,
+		TrustedDataValidators: []string{newAddr.String()},
+		Owner:                 acc1.String(),
+	}
+
+	dealId, err := suite.MarketKeeper.CreateNewDeal(suite.Ctx, acc1, tempDeal)
+	suite.Require().NoError(err)
+
+	testCert := makeTestCert()
+	_, err = suite.MarketKeeper.SellOwnData(suite.Ctx, acc3, testCert)
+	suite.Require().NoError(err)
+
+	findDeal, err := suite.MarketKeeper.GetDeal(suite.Ctx, dealId)
+	suite.Require().NoError(err)
+
+	dealAddress, err := types.AccDealAddressFromBech32(findDeal.GetDealAddress())
+	suite.Require().NoError(err)
+
+	dealBalance := suite.BankKeeper.GetBalance(suite.Ctx, dealAddress, assets.MicroMedDenom)
+
+	beforeDeactivateBalance := suite.BankKeeper.GetBalance(suite.Ctx, acc1, assets.MicroMedDenom)
+
+	deactivatedDealId, err := suite.MarketKeeper.DeactivateDeal(suite.Ctx, dealId, acc1)
+	suite.Require().NoError(err)
+
+	newAcc1Balance := suite.BankKeeper.GetBalance(suite.Ctx, acc1, assets.MicroMedDenom)
+
+	suite.Require().Equal(deactivatedDealId, dealId)
+
+	suite.Require().Equal(newAcc1Balance, beforeDeactivateBalance.Add(dealBalance))
+}
+
+func (suite *dealTestSuite) TestIsNotEqualOwner() {
+	err := suite.BankKeeper.AddCoins(suite.Ctx, acc1, sdk.NewCoins(sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(10000000000))))
+	suite.Require().NoError(err)
+
+	err = suite.BankKeeper.AddCoins(suite.Ctx, acc3, zeroFunds)
+	suite.Require().NoError(err)
+
+	tempDeal := types.Deal{
+		DataSchema:            []string{"http://jsonld.com"},
+		Budget:                &sdk.Coin{Denom: assets.MicroMedDenom, Amount: sdk.NewInt(1000000000)},
+		MaxNumData:            10000,
+		TrustedDataValidators: []string{newAddr.String()},
+		Owner:                 acc1.String(),
+	}
+
+	dealId, err := suite.MarketKeeper.CreateNewDeal(suite.Ctx, acc1, tempDeal)
+	suite.Require().NoError(err)
+
+	testCert := makeTestCert()
+	_, err = suite.MarketKeeper.SellOwnData(suite.Ctx, acc3, testCert)
+	suite.Require().NoError(err)
+
+	_, err = suite.MarketKeeper.DeactivateDeal(suite.Ctx, dealId, acc2)
+	suite.Require().Error(err, "the owner of deal and requester is not equal")
```

Comment:
Why don't we use sentinel error?
I think it'd be better to define and use sentinel error in both `deal.go` and `deal_test.go`.
When we edit the error message in this case, we have to edit this assertion string too.